pkg/client/client.go

```diff
@@ -284,12 +297,36 @@ type imageFactory struct {
 	keychain     authn.Keychain
 }
 
-func (f *imageFactory) NewImage(repoName string, daemon bool, imageOS string) (imgutil.Image, error) {
-	platform := imgutil.Platform{OS: imageOS}
+func (f *imageFactory) NewImage(repoName string, daemon bool, target dist.Target, multiarch bool) (imgutil.Image, error) {
+	// TODO Check the equivalent with the Target OS Version
```

Comment:
What does this mean?

Reply:
When I was writing this code, the `imgutil.Platform` struct didn't have `OS Version` I think

Reply:
Oh, no, I remembered now!

Our `Target` struct doesn't have `OS Version` defined. The `TODO` was to remember me if we can fetch an Image also using that attribute

```go
type Target struct {
	OS            string         `json:"os" toml:"os"`
	Arch          string         `json:"arch" toml:"arch"`
	ArchVariant   string         `json:"variant,omitempty" toml:"variant,omitempty"`
	Distributions []Distribution `json:"distros,omitempty" toml:"distros,omitempty"`
}
```

Reply:
Are we handling this TODO now?

Reply:
I just created this [PR](https://github.com/buildpacks/imgutil/pull/273/) on imgutil to add the missing `variant` attribute, once we have the merge, I think we will be ok with this

Reply:
I tried to fix this one on this [commit](https://github.com/buildpacks/pack/pull/2086/commits/88780df4827db9d324f81e13e67a7f70de4caf3f)